images/metering-ansible-operator/roles/meteringconfig/tasks/configure_metastore.yml

```diff
@@ -0,0 +1,33 @@
+- name: Get metastore secret
+  block:
+  - name: Get Metastore storage credentials secret
+    k8s_info:
+      api_version: v1
+      kind: Secret
+      name: "{{ _hive_metastore_db_secretName }}"
+      namespace: "{{ meta.namespace }}"
+    register: hive_metastore_db_secret_buf
+
+  - name: Create variables for the decoded username and password data
+    set_fact:
+      _hive_metastore_db_username: "{{ _hive_metastore_db_secret_buf_resources_first_index.data.username | b64decode }}"
+      _hive_metastore_db_password: "{{ _hive_metastore_db_secret_buf_resources_first_index.data.password | b64decode }}"
```

Comment:
We should add a validation check that the decoded username and password are not empty before merging with the `meteringconfig_spec` dictionary.

Reply:
I tried this in two ways.
One way was by adding conditionals to the task:
```
  - name: Create variables for the decoded username and password data
    set_fact:
      _hive_metastore_db_username: "{{ _hive_metastore_db_secret_buf_resources_first_index.data.username | b64decode }}"
      _hive_metastore_db_password: "{{ _hive_metastore_db_secret_buf_resources_first_index.data.password | b64decode }}"
      meteringconfig_spec: "{{ meteringconfig_spec | combine(_hive_metastore_db_overrides, recursive=True) }}"
    when:
    - _hive_metastore_db_username | length > 0
    - _hive_metastore_db_password | length > 0
    - hive_metastore_db_secret_buf is defined
    - _hive_metastore_db_secret_buf_resources_first_index | length > 0
```

This did not work.

The other way was by making additions to validate_hive_storage and adding assertions:
```
# 
#  Validate Hive Metastore Database secret is not empty
# 
 - name: Validate Hive Metastore Database secret configuration
   block:
   - name: Validate the user-provided Hive Database username is non-empty
     assert:
       that:
         - _hive_metastore_db_username != ""
       msg: "hive database secret username cannot be empty"

   - name: Validate the user-provied Hive Database password is non-empty
     assert:
       that:
         - _hive_metastore_db_password != ""
      msg: "hive database secret password cannot be empty"
```
Neither of these worked.
What did you add?